CanDIG/query
# DIG-1833: Query returns all donors to the clinical data table when no donors match a genomic query

# Jira Ticket

- [DIG-1833](https://candig.atlassian.net/browse/DIG-1833)

# Description
This fixes a bug where Query would return all donors authorized to a user when searching on a gene with no matching donors for the current user, but while there is a matching donor that the user is not authorized to view.

# Screenshots
[Screencast from 2024-10-24 10-57-40.webm](https://github.com/user-attachments/assets/5808140d-2dec-4f41-a130-11d3e79a625e)

# To test
1. `make recompose-query` with this branch
2. Log in as user2
3. Search SLX9
4. Nothing should show up, though a donor does exist for user1
5. Nothing else should have  changed -- when no matches exist or when logged in as the correct user it should still work.

[DIG-1833]: https://candig.atlassian.net/browse/DIG-1833?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
